how-to-guides/present-cross-validation-results/scripts/neptune_cross_validation.py

```diff
@@ -0,0 +1,118 @@
+import neptune.new as neptune
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from torchvision import datasets, transforms
+from torch.utils.data import SubsetRandomSampler, DataLoader
+from sklearn.model_selection import KFold
+from statistics import mean
+
+# Step 1: Create a Neptune Run
+run = neptune.init(
+    project="common/showroom",
+    api_token="ANONYMOUS",
+    tags="cross-validation",
+)
+
+device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+# Step 2: Log config and hyperparameters
+parameters = {
+    "epochs": 1,
+    "lr": 1e-2,
+    "bs": 10,
+    "input_sz": 32 * 32 * 3,
+    "n_classes": 10,
+    "k_folds": 2,
+    "model_name": "checkpoint.pth",
+    "seed": 42,
+}
+
+# Log hyperparameters
+run["global/parameters"] = parameters
+
+# Seed
+torch.manual_seed(parameters["seed"])
+
+# Model
+class BaseModel(nn.Module):
+    def __init__(self, input_sz, hidden_dim, n_classes):
+        super(BaseModel, self).__init__()
+        self.main = nn.Sequential(
+            nn.Linear(input_sz, hidden_dim * 2),
+            nn.ReLU(),
+            nn.Linear(hidden_dim * 2, hidden_dim),
+            nn.ReLU(),
+            nn.Linear(hidden_dim, hidden_dim // 2),
+            nn.ReLU(),
+            nn.Linear(hidden_dim // 2, n_classes),
+        )
+
+    def forward(self, input):
+        x = input.view(-1, 32 * 32 * 3)
+        return self.main(x)
+
+
+model = BaseModel(
+    parameters["input_sz"], parameters["input_sz"], parameters["n_classes"]
+).to(device)
+criterion = nn.CrossEntropyLoss()
+optimizer = optim.SGD(model.parameters(), lr=parameters["lr"])
+
+# trainset
+data_dir = "data/CIFAR10"
+compressed_ds = "./data/CIFAR10/cifar-10-python.tar.gz"
+data_tfms = {
+    "train": transforms.Compose(
+        [
+            transforms.RandomHorizontalFlip(),
+            transforms.ToTensor(),
+            transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]),
+        ]
+    )
+}
+trainset = datasets.CIFAR10(data_dir, transform=data_tfms["train"], download=True)
+dataset_size = len(trainset)
+
+run["global/dataset/CIFAR-10"].track_files(data_dir)
```

Comment:
We should use sth. else instead of "global",

How would you call it? Maybe "Training_results", or "Results"?

Reply:
Make sure to chage it also in notebook